src/main/java/com/google/firebase/messaging/ApnsFcmOptions.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.messaging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.Key;
+
+/**
+ * Represents the APNS-specific FCM options that can be included in an {@link ApnsConfig}. Instances
+ * of this class are thread-safe and immutable.
+ */
+public final class ApnsFcmOptions {
+
+  @Key("analytics_label")
+  private final String analyticsLabel;
+
+  private ApnsFcmOptions(Builder builder) {
+    FcmOptionsUtil.checkAnalyticsLabel(builder.analyticsLabel);
+    this.analyticsLabel = builder.analyticsLabel;
+  }
+
+  /**
+   * Creates a new {@link ApnsFcmOptions} with the specified analytics label.
+   *
+   * @param analyticsLabel An analytics label
+   * @return An ApnsFcmOptions with the analytics label set to the supplied value.
+   */
+  public static ApnsFcmOptions withAnalyticsLabel(String analyticsLabel) {
+    return new ApnsFcmOptions(builder().setAnalyticsLabel(analyticsLabel));
```

Comment:
Nit: This could be written just using the public APIs

```
return builder().setAnalyticsLabel(analyticsLabel).build();
```

Same in other similar places.

Reply:
done